nexus/tests/integration_tests/sp_updater.rs

```diff
@@ -0,0 +1,445 @@
+//! Nexus integration tests
+//!
```

Comment:
Out of curiosity, could/should these be unit tests rather than integration tests?

Reply:
Not sure; they definitely need all the MGS-with-simulated-SPs stuff, which I thought tipped them over into integration tests. Do we have unit tests that stand up that much supporting machinery?

Reply:
We do (see all the uses of `#[nexus_test]` in the unit tests -- that thing stands up all kinds of stuff).  It does make me wonder what the distinction means for us given that we're not really testing a library crate in the usual sense.  I definitely wouldn't worry too much about it.  I was just wondering.